src/freeze-dry/common.test.js

```diff
@@ -0,0 +1,97 @@
+        const dataUri = await urlToDataUri('example.com')
+        expect(spy).toHaveBeenCalled()
+        expect(dataUri).toBe('about:invalid')
+        spy.mockRestore()
+    })
+
+    test('should return a "about:invalid" for an error URL', async () => {
+        fetch.mockResponseOnce(new Error())
+        const dataUri = await urlToDataUri('example.com')
+        expect(dataUri).toBe('about:invalid')
+    })
+
+    test('should return the same url if already in dataUri form', async () => {
+        const dataUri = await urlToDataUri('data:;base64,test')
+        expect(dataUri).toBe('data:;base64,')
+    })
+})
+
+describe('inlineUrlsInAttributes', () => {
+    const docUrl = 'https://example.com/page'
+    const parser = new DOMParser()
+
+    test('should change the URL in <img> tag to a dataUri', async () => {
+        fetch.mockResponseOnce(new Blob([ {example: 'Lorem Ipsum'} ]))
+        expect.assertions(2)
+        const rootElement = parser.parseFromString(
+            '<html><body><img src="public/image/background.png" alt="background" /></body></html>',
+            'text/html'
+        )
+        await inlineUrlsInAttributes({elements: 'img', attributes: 'src', fixIntegrity: true, rootElement, docUrl})
+        expect(rootElement.querySelector('img').getAttribute('data-original-src')).toBe('public/image/background.png')
+        expect(rootElement.querySelector('img').getAttribute('src')).toBe('data:;base64,W29iamVjdCBPYmplY3Rd')
+    })
+
+    test('should change the URL in the <link> tag to a dataUri', async () => {
+        fetch.mockResponseOnce(new Blob([ {example: 'Lorem Ipsum'} ]))
+        expect.assertions(2)
+        const rootElement = parser.parseFromString(
+            '<html><head><link rel="icon" href="public/image/favicon.ico"></head></html>',
+            'text/html'
+        )
+        await inlineUrlsInAttributes({elements: 'link', attributes: 'href', fixIntegrity: true, rootElement, docUrl})
+        expect(rootElement.querySelector('link').getAttribute('data-original-href')).toBe('public/image/favicon.ico')
+        expect(rootElement.querySelector('link').getAttribute('href')).toBe('data:;base64,W29iamVjdCBPYmplY3Rd')
+    })
+
+    test('should change the URL in srcset of the <img> tag to a dataUri', async () => {
+        fetch.mockResponseOnce(new Blob([ {example: 'Lorem Ipsum'} ]))
+        expect.assertions(2)
+        const rootElement = parser.parseFromString(
+            '<html><body><img src="background.png" alt="background" srcset="background.png 200w,background.png 400w"></body></html>',
+            'text/html'
+        )
+        await inlineUrlsInAttributes({elements: 'img', attributes: 'srcset', fixIntegrity: true, rootElement, docUrl})
+        expect(rootElement.querySelector('img').getAttribute('data-original-srcset')).toBe('background.png 200w,background.png 400w')
+        expect(rootElement.querySelector('img').getAttribute('srcset')).toBe('data:;base64,W29iamVjdCBPYmplY3Rd')
+    })
```

Comment:
Same as above: no apparent added value. We could add a test for the attrToUrls option instead, which is used for attributes such as `srcset` with multiple URIs in them.